spec/tools/environment_builders/openstack/services/compute/builder.rb

```diff
@@ -64,9 +64,13 @@ def find_or_create_servers(volumes, networks, security_groups, images, networkin
                             }) if volume
             end
 
-            if (network_name = server.delete(:__network_name))
-              network = networks.detect { |x| x.name == network_name }
-              server.merge!(:nics => [{"net_id" => network.id}]) if network
+            if (network_names = server.delete(:__network_names))
+              nics = []
+              network_names.each do |network_name|
+                network = networks.detect { |x| x.name == network_name }
+                nics << {"net_id" => network.id} if network
+              end
```

Comment:
I probably over-engineering and micro-optimising here, but an alternative solution there could be to build a set of network names (O(n), where n = `network_names.size`), and then select all matching networks using this set (O(m), where m = `networks.size`). So the total cost would be O(n+m). Currently it is O(nm).

```
network_names_set = network_names.to_set
nics = networks
  .select { |network| network.name.in?(network_names_set) }
  .map { |network| { 'net_id' => network.id } }
```

As a bonus, the new solution doesn't use mutation (`<<`) and `if`s.